### ISSUE TITLE
Tensorflow & Keras update causes significantly worse model performance. 

### ISSUE THREAD

**Issue by hammad93** [tensorflow/tensorflow#42698]
<em>Please make sure that this is an issue related to performance of TensorFlow.
As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:performance_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

**Describe the expected behavior**

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

While doing research, I make sure to architect it to be reproducible and have quality assurance checks. When using Google Colab which has the most recent version of Tensorflow and Keras, I noticed that the loss was dramatically worse for the exact same code, as referenced here, https://github.com/hammad93/hurricane-net/issues/7

The solution was to downgrade Keras to 2.2 although I think 2.3 would work too. I'm not sure if it's the learning rate because even after increasing the epochs, I wasn't able to achieve the same accuracy. I won't be able to switch over to Tensorflow or the newest version of Keras. Just for reference, the model loss as reported by mean squared error is supposed to be 0.31 but the lowest I have been able to get with the exact same code is 0.40 and doing a gaussian hyper parameter search does not fix it. 

Here's the notebook for reference, https://github.com/hammad93/hurricane-net/blob/master/hurricane_net.ipynb

**Comment by amahendrakar** [comment]
@hammad93,
On running the code with TF v2.3 and Keras 2.4.3, I did get the loss value upto 0.3291. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/18ab5a3df44ef2d62bf391eaa887f15a/hurricane_net.ipynb#scrollTo=0ETjdPZJTA6H).
  
However, running the code with Keras 2.2 take much longer. Each epoch takes ~10s when compared to less than 0s on TF v2.3, take a look at [this gist](https://colab.research.google.com/gist/amahendrakar/df0b2de345afd8e96e4471335f31111a/tf-1-14-and-keras-2-2.ipynb) for reference. 

Also, the example you have provided is fairly complex. Can you remove the dependencies and get the example down to the simplest possible repro? So that it will be easier to debug the issue. Thanks!
